### PR TITLE
[Tests tidy] Broadcasts service 

### PR DESCRIPTION
**Pull request by tatitati**
Main changes:
- Added extra tests to test pagination on FindByVersionTest.php
- Removed unused function broadcastsFromDbData() (is not used anymore in order to get more visibility on tests)
- Modified explicit dates for dynamics "+/-1year"
- Simplified some tests

### CODE REVIEW

File: tests/Service/BroadcastsService/AbstractBroadcastsServiceTest.php [hunk 1]
```diff
@@ -2,31 +2,28 @@
...
-    public function setUp()
+    public function setUp(): void
     {
         $this->setUpCache();
         $this->setUpRepo('BroadcastRepository');
         $this->setUpMapper('BroadcastMapper', 'broadcastFromDbData');
     }
 
-    protected function broadcastsFromDbData(array $entities)
+    protected function broadcastFromDbData(array $dbData): Broadcast
```
Comment (votemike) [inline tests/Service/BroadcastsService/AbstractBroadcastsServiceTest.php]: Docblock for array?

Comment (votemike) [inline tests/Service/BroadcastsService/AbstractBroadcastsServiceTest.php]: Also `@return Broadcast|PHPUnit_Mock` or whatever the mock is.

File: tests/AbstractServiceTest.php [hunk 2]
```diff
@@ -40,7 +40,7 @@ protected function setUpMapper($mapperName, $entityBuilderMethod)
     {
         $this->mockMapper = $this->createMock($this::MAPPER_NS . $mapperName);
 
-        $this->mockMapper->expects($this->any())
+        $this->mockMapper
```
Comment (votemike) [inline tests/AbstractServiceTest.php]: Do we want this to be `->expects(1)`?

File: tests/Service/BroadcastsService/AbstractBroadcastsServiceTest.php [hunk 3]
```diff
@@ -2,31 +2,31 @@
...
+    public function setUp(): void
     {
         $this->setUpCache();
         $this->setUpRepo('BroadcastRepository');
         $this->setUpMapper('BroadcastMapper', 'broadcastFromDbData');
     }
 
-    protected function broadcastsFromDbData(array $entities)
+    /**
+     * @param array[] $dbData
```
Comment (votemike) [inline tests/Service/BroadcastsService/AbstractBroadcastsServiceTest.php]: This isn't an array of arrays is it? Shouldn't it be `mixed[]`?

File: tests/Service/BroadcastsService/AbstractBroadcastsServiceTest.php [hunk 4]
```diff
@@ -2,31 +2,31 @@
...
 {
-    public function setUp()
+    public function setUp(): void
     {
         $this->setUpCache();
         $this->setUpRepo('BroadcastRepository');
         $this->setUpMapper('BroadcastMapper', 'broadcastFromDbData');
     }
 
-    protected function broadcastsFromDbData(array $entities)
```
Comment (votemike) [inline tests/Service/BroadcastsService/AbstractBroadcastsServiceTest.php]: What about removing these types of methods from the other service tests?

File: tests/Service/BroadcastsService/FindByVersionTest.php [hunk 5]
```diff
@@ -2,49 +2,65 @@
...
+        $version = $this->mockEntity('Version', $dbId);
+        $dbData = [['pid' => 'b00swyx1'], ['pid' => 'b010t150']];
+
+        $this->mockRepository
+            ->method('findByVersion')
+            ->willReturn($dbData);
+
+        $broadcasts = $this->service()->findByVersion($version);
+
+        $this->assertCount(2, $broadcasts);
```
Comment (votemike) [inline tests/Service/BroadcastsService/FindByVersionTest.php]: 👍

File: tests/Service/BroadcastsService/FindByServiceAndDateRangeTest.php [hunk 6]
```diff
@@ -2,47 +2,66 @@
...
+        $broadcasts = $this->service()->findByServiceAndDateRange($sid, $fromDateTime, $toDatetime);
 
-        $this->assertEquals($this->broadcastsFromDbData($dbData), $broadcasts);
+        $this->assertCount(2, $broadcasts);
+        $this->assertContainsOnly(Broadcast::class, $broadcasts);
+        $this->assertEquals('b00swyx1', (string) $broadcasts[0]->getPid());
+        $this->assertEquals('b010t150', (string) $broadcasts[1]->getPid());
     }
 
     public function testFindByServiceAndDateRangeWhenNoBroadcastsFound()
```
Comment (votemike) [inline tests/Service/BroadcastsService/FindByServiceAndDateRangeTest.php]: I feel this test is simple/similar enough to put in with the pagination provider. (Maybe change the method name of paginationProvider). It just avoids repetition.

File: tests/AbstractServiceTest.php (hunk 2 above)
Comment (tatitati) [inline tests/AbstractServiceTest.php]: Just checked. is not consistent the amount of calls to getDomainModel. So Expects($this->once()) breaks many tests. jeje. I will manage anyway...maybe I can do something as I go with the tidy up.

File: tests/Service/BroadcastsService/AbstractBroadcastsServiceTest.php (hunk 3 above)
Comment (tatitati) [inline tests/Service/BroadcastsService/AbstractBroadcastsServiceTest.php]: well, actually is an array of arrays. arrays represent here "rows" in db, and each row is a mixed array with the fields/values

File: tests/Service/BroadcastsService/AbstractBroadcastsServiceTest.php (hunk 4 above)
Comment (tatitati) [inline tests/Service/BroadcastsService/AbstractBroadcastsServiceTest.php]: On my way, but I will go service by service. :smile:

File: tests/Service/BroadcastsService/FindByServiceAndDateRangeTest.php (hunk 6 above)
Comment (tatitati) [inline tests/Service/BroadcastsService/FindByServiceAndDateRangeTest.php]: arg.......I still think that this tests is testing something different of pagination. If this test fails one day, we will know quickly that pagination was working properly, and is related with this specific behaviour

File: tests/Service/BroadcastsService/AbstractBroadcastsServiceTest.php (hunk 3 above)
Comment (votemike) [inline tests/Service/BroadcastsService/AbstractBroadcastsServiceTest.php]: If it's an array of arrays, how can you access `$dbData['pid']`? Is the value of `$dbData['pid']` an array?

Comment (tatitati) [inline tests/Service/BroadcastsService/AbstractBroadcastsServiceTest.php]: true, didn't spot it. That function is called per row, so is the mixed[]

File: tests/Service/BroadcastsService/FindByServiceAndDateRangeTest.php [hunk 7]
```diff
@@ -2,48 +2,88 @@
...
+            ],
+        ];
+    }
 
-        $this->assertSame([], $broadcasts);
+    /**
+     * @param Broadcast[] $broadcasts
+     * @return string[]
+     */
+    private function extractPids(array $broadcasts): array
```
Comment (BPScott) [inline tests/Service/BroadcastsService/FindByServiceAndDateRangeTest.php]: We're going to use this in pretty much all services as they follow a similar pattern right? Would it be better placed in the AbstractServiceTest so everything can use it

Comment (tatitati) [inline tests/Service/BroadcastsService/FindByServiceAndDateRangeTest.php]: :+1:

File: tests/Service/BroadcastsService/FindByServiceAndDateRangeTest.php [hunk 8]
```diff
@@ -37,7 +37,7 @@ public function paginationProvider()
             'custom pagination' => [3, 12, [3, 5]],
         ];
     }
-
+
```
Comment (votemike) [inline tests/Service/BroadcastsService/FindByServiceAndDateRangeTest.php]: Why?

File: tests/AbstractServiceTest.php [hunk 9]
```diff
@@ -36,15 +36,18 @@ protected function setUpCache()
             ->getMock();
     }
 
-    protected function setUpMapper($mapperName, $entityBuilderMethod)
+    protected function setUpMapper($mapperName, $callback)
```
Comment (votemike) [inline tests/AbstractServiceTest.php]: Typehint string and callable?

Comment (BPScott) [inline tests/AbstractServiceTest.php]: The seccond arg is currently callable|string as we haven't retrofitted the rest of the service tests yet.

**Pull request merged**